com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/Editor/AssetPostProcessors/MaterialPostprocessor.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Rendering.BuiltIn;
+using UnityEngine;
+using static Unity.Rendering.BuiltIn.ShaderUtils;
+
+namespace UnityEditor.Rendering.BuiltIn
+{
+    class MaterialModificationProcessor : AssetModificationProcessor
+    {
+        static void OnWillCreateAsset(string asset)
+        {
+            if (!asset.ToLowerInvariant().EndsWith(".mat"))
+            {
+                return;
+            }
+            MaterialPostprocessor.s_CreatedAssets.Add(asset);
+        }
+    }
+
+    class MaterialReimporter : Editor
+    {
+        static bool s_NeedToCheckProjSettingExistence = true;
+
+        static void ReimportAllMaterials()
+        {
+            string[] guids = AssetDatabase.FindAssets("t:material", null);
+            // There can be several materials subAssets per guid ( ie : FBX files ), remove duplicate guids.
+            var distinctGuids = guids.Distinct();
+
+            int materialIdx = 0;
+            int totalMaterials = distinctGuids.Count();
+            foreach (var asset in distinctGuids)
+            {
+                materialIdx++;
+                var path = AssetDatabase.GUIDToAssetPath(asset);
+                EditorUtility.DisplayProgressBar("Material Upgrader re-import", string.Format("({0} of {1}) {2}", materialIdx, totalMaterials, path), (float)materialIdx / (float)totalMaterials);
+                AssetDatabase.ImportAsset(path);
+            }
+            EditorUtility.ClearProgressBar();
+
+            MaterialPostprocessor.s_NeedsSavingAssets = true;
+        }
+
+        [InitializeOnLoadMethod]
+        static void RegisterUpgraderReimport()
+        {
+
+        }
+    }
+
+    class MaterialPostprocessor : AssetPostprocessor
```

Comment:
I cut out all of the versioning logic here since this is V0 and no upgrading is necessary. There's a decent enough chance this is wrong so I could use some extra eyes here. In particular I'm also curious what's preventing this from running on URP materials but haven't dug in yet.

Reply:
We get to assume at least 2021.2 for versioning purposes.